From 893505fe557db3fbfdb0e2bf74a5030a2f610bd8 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Wed, 14 Mar 2018 11:07:42 +0800
Subject: [PATCH 3486/5242] MLK-17802 drm/imx: dpu: plane: Remove pixel
 formats unsupported by DPR

commit  87c89950e782b66bf411654da9d9c6d13565f442 from
https://source.codeaurora.org/external/imx/linux-imx.git

The RGB888/BGR888/NV16/NV61/NV24/NV42 pixel formats are not supported
by DPR.  They cannot get the benefits(i.e., tile resolving and
underrun-proof) from the prefetch engines.  Also, 16bit and 32bit
RGB pixel formats are widely used by GUI, while NV12 pixel format
is the only pixel format supported by VPU.  Thus, it makes little
sense to support the pixel formats which are not supported by DPR.
Another idea is that removing the pixel formats makes the driver
a bit simpler since we don't have to deal with the cases in which
prefetch engines are bypassed.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dpu/dpu-plane.c |   20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/drivers/gpu/drm/imx/dpu/dpu-plane.c b/drivers/gpu/drm/imx/dpu/dpu-plane.c
index 7b676fde..8a3aeab 100644
--- a/drivers/gpu/drm/imx/dpu/dpu-plane.c
+++ b/drivers/gpu/drm/imx/dpu/dpu-plane.c
@@ -35,18 +35,12 @@
 	DRM_FORMAT_RGBX8888,
 	/* DRM_FORMAT_BGRA8888, */
 	DRM_FORMAT_BGRX8888,
-	DRM_FORMAT_RGB888,
-	DRM_FORMAT_BGR888,
 	DRM_FORMAT_RGB565,
 
 	DRM_FORMAT_YUYV,
 	DRM_FORMAT_UYVY,
 	DRM_FORMAT_NV12,
 	DRM_FORMAT_NV21,
-	DRM_FORMAT_NV16,
-	DRM_FORMAT_NV61,
-	DRM_FORMAT_NV24,
-	DRM_FORMAT_NV42,
 };
 
 static const uint64_t dpu_format_modifiers[] = {
@@ -130,14 +124,8 @@ static bool dpu_drm_plane_format_mod_supported(struct drm_plane *plane,
 		return false;
 
 	switch (format) {
-	case DRM_FORMAT_RGB888:
-	case DRM_FORMAT_BGR888:
 	case DRM_FORMAT_YUYV:
 	case DRM_FORMAT_UYVY:
-	case DRM_FORMAT_NV16:
-	case DRM_FORMAT_NV61:
-	case DRM_FORMAT_NV24:
-	case DRM_FORMAT_NV42:
 		return modifier == DRM_FORMAT_MOD_LINEAR;
 	case DRM_FORMAT_XRGB8888:
 	case DRM_FORMAT_XBGR8888:
@@ -359,10 +347,6 @@ static int dpu_plane_atomic_check(struct drm_plane *plane,
 		break;
 	case DRM_FORMAT_NV12:
 	case DRM_FORMAT_NV21:
-	case DRM_FORMAT_NV16:
-	case DRM_FORMAT_NV61:
-	case DRM_FORMAT_NV24:
-	case DRM_FORMAT_NV42:
 		bpp = 8;
 		break;
 	default:
@@ -489,10 +473,6 @@ static void dpu_plane_atomic_update(struct drm_plane *plane,
 		break;
 	case DRM_FORMAT_NV12:
 	case DRM_FORMAT_NV21:
-	case DRM_FORMAT_NV16:
-	case DRM_FORMAT_NV61:
-	case DRM_FORMAT_NV24:
-	case DRM_FORMAT_NV42:
 		bpp = 8;
 		break;
 	default:
-- 
1.7.9.5

